clang/lib/CodeGen/CoverageMappingGen.cpp

```diff
@@ -1503,38 +1527,53 @@ struct CounterCoverageMappingBuilder
     }
 
     // Create Branch Region around condition.
-    createBranchRegion(S->getCond(), BodyCount,
-                       subtractCounters(CondCount, BodyCount));
+    if (!llvm::EnableSingleByteCoverage)
+      createBranchRegion(S->getCond(), BodyCount,
+                         subtractCounters(CondCount, BodyCount));
```

Comment:
As far as I have checked, I guess `!EnableSingleByteCoverage` can be sunk into `createBranchRegion()` rather than checking from callers.

Sorry for my delayed comment. I didn't notice till merging.